resources/util.ts

```diff
@@ -77,6 +79,84 @@ const jobToRoleMap: Map<Job, Role> = (() => {
   return map;
 })();
 
+type WatchCombatantParams = {
+  ids?: number[];
+  names?: string[];
+  props?: string[];
+  delay?: number;
+  maxRetries?: number;
+  maxDuration?: number;
```

Comment:
What's the use case for both of these? It seems like they are saying the same thing in different units.

Reply:
maxRetries doesn't account for execution time of the overlay call, so if the call takes 50ms, after 20 retries that's a full second's worth of skew. I'd think just `maxDuration` should be sufficient but I'm not sure what use case @Trim21 was thinking of when they suggested adding retries, so I opted to add both options.

Reply:
Ah, I see how they're not exactly equivalent.  It still seems like solving roughly the same problem.  Maybe `maxDuration` is the more conservative estimate and that on its own is good enough?  I can't quite imagine why somebody would need `maxRetries` precisely.